tutorials/xr/setting_up_xr.rst

```diff
@@ -0,0 +1,107 @@
+.. _doc_setting_up_xr:
+
+Setting up XR
+=============
+
+Introduction to the XR system in Godot
+--------------------------------------
+
+Godot provides a moduler XR system that allows different XR platforms to be implemented in such a way that many of the platform specifics are hidden away from the user.
+At the core sits the :ref:`XRServer <class_xrserver>` which acts as a central interface to the XR system that allows users to discover interfaces and interact with the components of the XR system.
+
+Each supported XR platform is implemented as an :ref:`XRInterface <class_xrinterface>`. Supported interfaces register themselves with the :ref:`XRServer <class_xrserver>` and can be queried with the ``find_interface`` method on the :ref:`XRServer <class_xrserver>`. When the desired interface is found it can be initialised by calling ``initialize`` on the interface. 
+
+.. warning::
+	A registered interface means nothing more then that the interface is available, if the interface is not supported by the host system initialization may fail and return ``false``. This can have many reasons and sadly the reasons differ from platform to platform. It can be because the user hasn't installed the required software, or that the user simply hasn't plugged in their headset. You as a developer must thus react properly on an interface failing to initialize. 
+
+Due to the special requirements for output in XR, especially for head mounted devices that supply different images to each eye, the :ref:`XRServer <class_xrserver>` in Godot will override various features in the rendering system. For stand alone devices this means the final output is handled by the :ref:`XRInterface <class_xrinterface>` and Godots normal output system is disabled. For desktop XR devices that work as a second screen it is possible to dedicate a separate :ref:`Viewport <class_viewport>` to handle the XR output leaving the main Godot window available for displaying alternative content.
+
+.. note::
+	Note that only one interface can be responsible for handling the output to an XR device, this is known as the primary interface and by default will be the first interface that is initialized. Godot currently thus only supports implementations with a single headset.
+	It is possible, but increasingly uncommon, to have a secondary interface such as an interface solely responsible for adding tracking to an otherwise 3DOF only device.
```

Comment:
```suggestion
	It is possible, but increasingly uncommon, to have a secondary interface, for example to add tracking to an otherwise 3DOF only device.
```